File: 9041 - Software Instruction/test07/distinct_lines.pl
#!/usr/bin/perl -w

my $number = $ARGV[0];
my @uniq = [];
my $count = 0;

while ($line=<STDIN>){

	$count++;
	chomp($line);
	$line = lc($line);
	$line =~ s/\s+/ /g;
	$line =~ s/^\s+|\s+$//g;
	#print "$line\n";
	if (!($line ~~ @uniq)){
		push @uniq, $line;
	}

	if($#uniq == $number ){
		#$count -= 1; 
		print "$#uniq distinct lines seen after $count lines read.\n";
		last;
	}

}

if ($#uniq < $number)
{
#	print "@uniq \n";
	print "End of input reached after $count lines read - $number different lines not seen.\n"
}

File: 9041 - Software Instruction/test09/distinct_lines.py
#!/usr/bin/python

import sys

number = sys.argv[1]

setLine = set()
finalNumber = 0

for line in sys.stdin.readlines():
	line = line.lower()
	array = line.split()
	line = ' '.join(array)
	setLine.add(line)
	finalNumber += 1
	if int(number) == len(setLine):
		break

if int(number) > len(setLine):
	print("End of input reached after {} lines read -  {} different lines not seen.".format(finalNumber,number))
else:
	print("{} distinct lines seen after {} lines read.".format(len(setLine),finalNumber))
